src/filesystem/implementation.js

```diff
@@ -533,7 +533,10 @@ function remove_directory(context, path, callback) {
   find_node(context, parentPath, read_parent_directory_data);
 }
 
-function open_file(context, path, flags, callback) {
+function open_file(context, path, flags, mode, callback) {
+  if (typeof mode == 'function'){
+    callback = mode;
```

Comment:
You should also reset `mode` here, since it *was* a `function`, and now you're moving that over to `callback.  You should set it to `null`.